cawemo-to-web-modeler-migration/migrate-cawemo-to-web-modeler.sh

```diff
@@ -0,0 +1,210 @@
+
+                NEW_FOLDER=$(cat .curl_tmp)
+
+                if [ "$HTTP_STATUS" != "200" ]
+                then
+                    echo "⚠ POST https://modeler.cloud.camunda.io/api/beta/folders failed with status code $HTTP_STATUS"
+                    echo "[$(date +%s)] ⚠ POST https://modeler.cloud.camunda.io/api/beta/folders failed with status code $HTTP_STATUS" >> migration.log
+                else
+                    NEW_FOLDER_ID=$(jq -r .id <<< "$NEW_FOLDER")
+
+                    folderIds[$OLD_FOLDER_ID]=$NEW_FOLDER_ID
+
+                    PARENT_ID="\"$NEW_FOLDER_ID\""
+
+                    echo "[$(date +%s)] Migrated folder $FOLDER_NAME (ID: $OLD_FOLDER_ID)" >> migration.log
+                    echo "\e[1A\e[K├── ✔ Migrated folder ${GREEN}$FOLDER_NAME${NC} (ID: $OLD_FOLDER_ID)."
+                fi
+            else
+                # Existing folder
+                PARENT_ID="\"${folderIds[$OLD_FOLDER_ID]}\""
+            fi
+
+            FILE_TREE_SPACES="    $FILE_TREE_SPACES"
+        done
+
+        if [ "$FILE_TYPE" = "TEMPLATE_GENERIC" ] || [ "$FILE_TYPE" = "TEMPLATE_SERVICE_TASK" ]
+        then
+            echo "$FILE_TREE_SPACES$TREE_SYMBOL ⚠ File ${GREEN}$FILE_NAME${NC} can't be migrated: A C7 $FILE_TYPE is not supported in C8. (ID: $OLD_FILE_ID)"
+            echo "[$(date +%s)] File $FILE_NAME can't be migrated: A C7 $FILE_TYPE is not supported in C8. (ID: $OLD_FILE_ID)" >> migration.log
+        else
+            echo "$FILE_TREE_SPACES$TREE_SYMBOL Migrating file ${GREEN}$FILE_NAME${NC} (Type: $FILE_TYPE, ID: $OLD_FILE_ID)..."
+
+            FILE_WITH_CONTENT=$(curl -s -u $CAWEMO_USER_ID:$CAWEMO_API_KEY "https://cawemo.com/api/v1/files/$OLD_FILE_ID")
+
+            CONTENT=$(jq -r .content <<< "$FILE_WITH_CONTENT")
+
+            CONTENT=$(sed 's/"/\\"/g' <<< $CONTENT)
+
+            CONTENT=$(tr -d '\n' <<< $CONTENT)
+
+            FILE_REQUEST="{
+                \"name\": \"$FILE_NAME\",
+                \"projectId\": \"$NEW_PROJECT_ID\",
+                \"parentId\": $PARENT_ID,
+                \"content\": \"$CONTENT\",
+                \"fileType\": \"$FILE_TYPE\"
+            }"
+
+            HTTP_STATUS=$(curl -s -o .curl_tmp -w "%{http_code}" -X POST "https://modeler.cloud.camunda.io/api/beta/files" \
+                -H "Content-Type: application/json" \
+                -H "Authorization: Bearer $TOKEN" \
+                -d "$FILE_REQUEST")
+
+            NEW_FILE=$(cat .curl_tmp)
+
+            if [ "$HTTP_STATUS" != "200" ]
+            then
+                echo "⚠ POST https://modeler.cloud.camunda.io/api/beta/files failed with status code $HTTP_STATUS"
+                echo "[$(date +%s)] ⚠ POST https://modeler.cloud.camunda.io/api/beta/files failed with status code $HTTP_STATUS" >> migration.log
+            else
+                echo "\e[1A\e[K$FILE_TREE_SPACES$TREE_SYMBOL ✔ Migrated file ${GREEN}$FILE_NAME${NC} (Type: $FILE_TYPE, ID: $OLD_FILE_ID)."
```

Comment:
The character is not displayed correctly:
```
\e[1A\e[K├── ✔ Migrated file New BPMN Diagram (Type: BPMN, ID: 561e690c-4424-4b9e-a483-bdbec98e8176).
```
Probably related to this shellcheck warning (see above):
`SC2028 (info): echo may not expand escape sequences. Use printf`

Only happens when I run the script directly with `./migrate-cawemo-to-web-modeler.sh`, not with `zsh migrate-cawemo-to-web-modeler.sh` 🤔.

Reply:
Works now when using `#!/bin/zsh`

@jfriedenstab please verify